test/config/utility.cc

```diff
@@ -1211,6 +1212,49 @@ bool ConfigHelper::setListenerAccessLog(const std::string& filename, absl::strin
   return true;
 }
 
+void ConfigHelper::initializeTlsKeyLog(
+    envoy::extensions::transport_sockets::tls::v3::CommonTlsContext& tls_context,
+    bool keylog_local_filter, bool keylog_remote_filter, bool keylog_negative,
+    const Network::Address::IpVersion version) {
+  if (!keylog_local_filter && !keylog_local_filter) {
+    return;
+  }
+  auto tls_keylog_path = tls_context.mutable_tls_keylog()->mutable_logfile_path();
+  if (!keylog_negative) {
+    *tls_keylog_path = TestEnvironment::temporaryPath("tlskey.log");
+  } else {
+    *tls_keylog_path = TestEnvironment::temporaryPath("tlskey-negative.log");
+  }
+  if (keylog_local_filter) {
+    auto tls_keylog_local = tls_context.mutable_tls_keylog()->mutable_local_address_range();
+    auto new_element_local = tls_keylog_local->Add();
+    if (!keylog_negative) {
+      new_element_local->set_address_prefix(Network::Test::getLoopbackAddressString(version));
+    } else {
+      if (version == Network::Address::IpVersion::v6) {
+        new_element_local->set_address_prefix("1::2");
+      } else {
+        new_element_local->set_address_prefix("127.0.0.2");
+      }
+    }
+    new_element_local->mutable_prefix_len()->set_value(32);
```

Comment:
This is the wrong prefix length for ipv6. I think you want 128 in that case. Same below for remote.

Reply:
will fix